lms/resources/_js_config.py

```diff
@@ -94,9 +43,20 @@ def add_document_url(self, document_url):
         :raise HTTPBadRequest: if a request param needed to generate the config
             is missing
         """
-        self.config["urls"]["via_url"] = via_url(self._request, document_url)
+        self._config["urls"]["via_url"] = via_url(self._request, document_url)
         self._add_canvas_submission_params(document_url=document_url)
 
+    def asdict(self):
```

Comment:
Why isn't this `as_dict()`?

Reply:
`asdict()` rather than `as_dict` is used in the standard library so I figured I'd follow that, e.g. https://docs.python.org/3/library/dataclasses.html?highlight=asdict#dataclasses.asdict and https://docs.python.org/3/library/collections.html?highlight=asdict#collections.somenamedtuple._asdict